include/translated-post.php

```diff
@@ -3,123 +3,219 @@
+			}
+
+			if ( empty( $this->model->options['media_support'] ) ) {
+				// In case the post type attachment is stored in the option.
+				unset( $post_types['attachment'] );
+			} else {
+				$post_types['attachment'] = 'attachment';
+			}
+
+			/**
+			 * Filters the list of post types available for translation.
+			 * The default are post types which have the parameter ‘public’ set to true.
+			 * The filter must be added soon in the WordPress loading process:
+			 * in a function hooked to ‘plugins_loaded’ or directly in functions.php for themes.
+			 *
+			 * @since 0.8
+			 *
+			 * @param string[] $post_types  List of post type names (as array keys and values).
+			 * @param bool     $is_settings True when displaying the list of custom post types in Polylang settings.
+			 */
+			$post_types = apply_filters( 'pll_get_post_types', $post_types, false );
+
+			if ( did_action( 'after_setup_theme' ) ) {
+				$this->model->cache->set( 'post_types', $post_types );
+			}
 		}
 
-		$lang = $this->get_object_term( $post_id, $this->tax_language );
-		return ! empty( $lang ) ? $this->model->get_language( $lang ) : false;
+		/** @var array<non-empty-string, non-empty-string> $post_types */
+		return $filter ? array_intersect( $post_types, get_post_types() ) : $post_types;
 	}
 
 	/**
-	 * Deletes a translation.
+	 * Returns true if Polylang manages languages for this object type.
 	 *
-	 * @since 0.5
+	 * @since 3.4
 	 *
-	 * @param int $id Post id.
-	 * @return void
+	 * @param string|string[] $object_type Object type (post type) name or array of object type names.
+	 * @return bool
+	 *
+	 * @phpstan-param non-empty-string|non-empty-string[] $object_type
 	 */
-	public function delete_translation( $id ) {
-		$id = $this->sanitize_int_id( $id );
-
-		if ( empty( $id ) ) {
-			return;
-		}
-
-		parent::delete_translation( $id );
-		wp_set_object_terms( $id, array(), $this->tax_translations );
+	public function is_translated_object_type( $object_type ) {
+		$post_types = $this->get_translated_object_types( false );
+		return ( is_array( $object_type ) && array_intersect( $object_type, $post_types ) || in_array( $object_type, $post_types ) || 'any' === $object_type && ! empty( $post_types ) );
 	}
 
 	/**
-	 * A join clause to add to sql queries when filtering by language is needed directly in query
+	 * Returns the IDs of the objects without language.
 	 *
-	 * @since 1.2
+	 * @since 3.4
 	 *
-	 * @param string $alias Alias for $wpdb->posts table
-	 * @return string join clause
+	 * @param int $limit Max number of objects to return. `-1` to return all of them.
+	 * @return int[] Array of object IDs.
+	 *
+	 * @phpstan-param -1|positive-int $limit
+	 * @phpstan-return list<positive-int>
 	 */
-	public function join_clause( $alias = '' ) {
-		global $wpdb;
-		if ( empty( $alias ) ) {
-			$alias = $wpdb->posts;
+	public function get_objects_with_no_lang( $limit ) {
```

Comment:
This is what I suggested in the PR's description:
>`get_objects_with_no_lang()` has been made generic and moved from `PLL_Translated_Term` to `PLL_Translatable_Object`. **`PLL_Translated_Post::get_objects_with_no_lang()` has been kept so far, but I think it could benefit from the generic version in term of performance.**

Reply:
Removed in [06c9436](https://github.com/polylang/polylang/pull/1159/commits/06c94369eece6e872ad0cf6113b820d0a061f60f), I can't test for now but I don't think it will cause any issue.